src/main/scala/centaur/Test.scala

```diff
@@ -119,52 +120,62 @@ object Operations {
-        val lastCallAttempt = allCallAttempts.elements.last.asJsObject
-        val filteredCallAttributes = lastCallAttempt.fields collect {
-          case (k2, callAttributes) if listOfkeys.contains(k2) => k2 -> callAttributes
-        }
-        k1 -> JsObject(filteredCallAttributes)
+  private def makeMetadataMap(metadataJson: String): Map[String, JsValue] = {
 
-      case(k, v) => throw new Exception(s"Wrong Js parameter for $k, Call attempts are expected to be in a JsArray")
-    }
+    val flattenedMetadataMap = metadataJson.parseJson.asJsObject.flatten().fields map { case(k, v) => k -> v }
+    val listOfKeys = Set("runtimeAttributes", "preemptible", "executionStatus")
+    val filteredMap = flattenedMetadataMap filter { case (k, v) => listOfKeys exists { x => k.contains(x) } }
 
-    lastCallAttempt
+    filteredMap
   }
 
   private def makeOutputMap(metadataJson: String): Map[String, JsValue] = {
     // Need DefaultJsonProtocol export to use convertTo[Map[String,String]]
     import DefaultJsonProtocol._
     val outputs = metadataJson.parseJson.asJsObject.fields.get("outputs").get.asJsObject.toString
-
     outputs.parseJson.convertTo[Map[String, JsValue]]
   }
 
+  /**
+    * This method accepts the two maps being compared, Workflow Request, and type of Map
+    * (metadata/output). It finds and returns the differences in the keys/values of the two maps.
+    */
+  def assertMapDiff(expectedMetadata: Option[Map[String, JsValue]], actualMetadata: Map[String, JsValue], request: WorkflowRequest, testType: String) = {
+    val expectedMap = expectedMetadata.get
+
+    val diff = expectedMap.keySet -- actualMetadata.keySet
+    println(s"For workflow ${request.name}: \nMissing expected $testType fields: $diff")
+
+    expectedMap foreach { case (k, v:JsValue) =>
+      if (actualMetadata.contains(k) && (!v.toString.equals(actualMetadata.get(k).mkString))) {
+          println(s"Unexpected $testType for key: $k. Found: ${v.toString} Expected: ${actualMetadata.get(k).mkString}")
+      }
+    }
+  }
 
   def verifyMetadataAndOutputs(workflow: Workflow, request: WorkflowRequest): Test[Workflow] = {
     new Test[Workflow] {
-      val expectedMap: Option[Map[String, JsObject]] = request.metadata map { metadataString: String => convertMetadataToTestMap(metadataString) }
+      val expectedMap: Option[Map[String, JsValue]] = request.metadata map { metadataString: String => makeMetadataMap(metadataString) }
       val expectedOutputMap: Option[Map[String, JsValue]] = request.metadata map { metadataString: String => makeOutputMap(metadataString) }
 
-      def verifyWorkflowMetadata(metadata: Map[String, JsObject]) = {
-        expectedMap match {
-          case Some(expected) if !expected.equals(metadata) => throw new Exception(s"Bad Metadata for Workflow ${request.name}")
-          case _ =>
-        }
+
+      def verifyWorkflowMetadata(metadata: Map[String, JsValue]) = {
+          expectedMap match {
```

Comment:
In this instance let's switch to the `if` - we can always pop back out to something richer if the time comes